### DISCUSSION TITLE
What does this parameter "perturbation_set" in function "initialize_model" mean?

### DISCUSSION THREAD

**Discussion by Winston8611** [meta-pytorch/botorch#2361] [Q&A] [answered]
In the "Fit the model" code block, the parameter "perturbation_set" in function "initialize_model" is an input parameter in the "FixedNoiseGP" function, which aims to add a certain amount of noise to the input x. However, I do not know the specific fitting process of this Gaussian process. What does it mean to add this noise?
As far as I understand it, the function "FixedNoiseGP" only needs to be trained with the input X and the corresponding Y. After adding noise to X, is the trained GP model used to predict these added noise X? But in this case, it is entirely possible to predict after training, so that the code logic is easier to understand, why add noise during training? But if not as I understand it, what is the purpose of adding noise to the input variable here?

The code is here:

`import torch 
def initialize_model(train_x, train_y, perturbation_set):`
    
> Constructs the model and its MLL.
    Args:
        train_x: An `n x d`-dim tensor of training inputs.
        train_y: An `n x m`-dim tensor of training outcomes.
        perturbation_set: A `n_w x d`-dim tensor of perturbations
    Returns:
        The MLL and the model. Note: the model is not trained!

    train_Yvar = torch.full_like(train_y, 1e-7) * train_y.std(dim=0).pow(2)
    models = []
    for i in range(train_y.shape[-1]):
        models.append(
            FixedNoiseGP(
                train_X=train_x,
                train_Y=train_y[..., i : i + 1],
                train_Yvar=train_Yvar[..., i : i + 1],
                outcome_transform=Standardize(m=1),
                input_transform=InputPerturbation(perturbation_set=perturbation_set),
            )
        )
    model = ModelListGP(*models)
    mll = SumMarginalLogLikelihood(model.likelihood, model)
    return mll, model

> An example of fitting a model.

    mll, model = initialize_model(
        train_x=X, train_y=Y, perturbation_set=perturbation_set )

**Comment by saitcakmak** [comment, marked as answer]
Hi @wenxiangj. I am guessing you're going through this tutorial: https://botorch.org/tutorials/robust_multi_objective_bo. BoTorch input transforms have flags that determine when they are applied. For `InputPerturbation`, the flag `transform_on_train` is set to `False` by default, which means the transform will not have any effect during model training. The flag `transform_on_eval` is set to `True`, so that during `posterior` evaluation, the noise is applied and the model is evaluated with all perturbations. The acquisition functions would then reduce these perturbations using a risk measure objective and optimize over that.
